MigrateServicePreferenceFromLegacy/handler.ts

```diff
@@ -0,0 +1,136 @@
+import { Context } from "@azure/functions";
+import { BlockedInboxOrChannelEnum } from "@pagopa/io-functions-commons/dist/generated/definitions/BlockedInboxOrChannel";
+import { RetrievedProfile } from "@pagopa/io-functions-commons/dist/src/models/profile";
+import {
+  makeServicesPreferencesDocumentId,
+  NewServicePreference,
+  ServicesPreferencesModel
+} from "@pagopa/io-functions-commons/dist/src/models/service_preference";
+import {
+  CosmosErrorResponse,
+  CosmosErrors
+} from "@pagopa/io-functions-commons/dist/src/utils/cosmosdb_model";
+import { NonNegativeInteger } from "@pagopa/io-functions-commons/node_modules/@pagopa/ts-commons/lib/numbers";
+import * as a from "fp-ts/lib/Array";
+import * as o from "fp-ts/lib/Option";
+import * as te from "fp-ts/lib/TaskEither";
+import * as t from "io-ts";
+import { FiscalCode } from "../generated/backend/FiscalCode";
+import { ServiceId } from "../generated/backend/ServiceId";
+import { errorsToError } from "../utils/conversions";
+
+const COSMOS_ERROR_KIND = "COSMOS_ERROR_RESPONSE";
+const CONFLICT_CODE = 409;
+const LOG_PREFIX = "MigrateServicePreferenceFromLegacy";
+
+export const MigrateServicesPreferencesQueueMessage = t.interface({
+  newProfile: RetrievedProfile,
+  oldProfile: RetrievedProfile
+});
+export type MigrateServicesPreferencesQueueMessage = t.TypeOf<
+  typeof MigrateServicesPreferencesQueueMessage
+>;
+
+function isCosmosError(
+  ce: CosmosErrors
+): ce is ReturnType<typeof CosmosErrorResponse> {
+  return ce.kind === COSMOS_ERROR_KIND;
+}
+
+export const createServicePreference = (
+  serviceId: ServiceId,
+  blockedChannels: ReadonlyArray<BlockedInboxOrChannelEnum>,
+  fiscalCode: FiscalCode,
+  version: NonNegativeInteger
+): NewServicePreference => ({
+  fiscalCode,
+  id: makeServicesPreferencesDocumentId(fiscalCode, serviceId, version),
+  isEmailEnabled: !blockedChannels.some(
+    channel => channel === BlockedInboxOrChannelEnum.EMAIL
+  ),
+  isInboxEnabled: !blockedChannels.some(
+    channel => channel === BlockedInboxOrChannelEnum.INBOX
+  ),
+  isWebhookEnabled: !blockedChannels.some(
+    channel => channel === BlockedInboxOrChannelEnum.WEBHOOK
+  ),
+  kind: "INewServicePreference",
+  serviceId,
+  settingsVersion: version
+});
+
+export const blockedsToServicesPreferences = (
+  blocked: {
+    [x: string]: readonly BlockedInboxOrChannelEnum[];
+  },
+  fiscalCode: FiscalCode,
+  version: NonNegativeInteger
+) =>
+  o
+    .fromNullable(blocked)
+    .map(b =>
+      Object.entries(b)
+        .filter(([serviceId, _]) => ServiceId.is(serviceId))
+        .map(
+          ([serviceId, blockedInboxOrChannelsForService]) =>
+            createServicePreference(
+              serviceId as ServiceId,
+              blockedInboxOrChannelsForService,
+              fiscalCode,
+              version
+            ) // cast required: ts do not identify filter as a guard
+        )
```

Comment:
`Array.filter` needs explicit definition to work as type guard
```suggestion
        .filter((_): _ is [ServiceId, BlockedInboxOrChannelEnum[]] =>
          ServiceId.is(_[0])
        )
        .map(([serviceId, blockedInboxOrChannelsForService]) =>
          createServicePreference(
            serviceId,
            blockedInboxOrChannelsForService,
            fiscalCode,
            version
          )
        )
```